master/internal/rm/kubernetesrm/spec.go

```diff
@@ -557,6 +560,33 @@ func configureSecurityContext(agentUserGroup *model.AgentUserGroup) *k8sV1.Secur
 	return nil
 }
 
+func getDetContainerSecurityContext(
+	agentUserGroup *model.AgentUserGroup,
+	env expconf.EnvironmentConfigV0,
```

Comment:
this isn't a hard and fast rule, since this is just a helper function without any real abstraction design or anything, but it would be nice to pass in just the subset of the `env` that is needed:

```
func getDetContainerSecurityContext(agentUserGroup *model.AgentUserGroup, pod *v1.Pod) {
    ...
}

...

SecurityContext: getDetContainerSecurityContext(spec.AgentUserGroup, env.PodSpec())
```

It's nice because:
- the typing makes it clearer what the helper function cares about
- you only have to type `env.PodSpec()` once when calling it rather than multiple times inside of it.

Reply:
Good idea!